src/webgpu/api/validation/capability_checks/limits/limit_utils.ts

```diff
@@ -1005,6 +1005,29 @@ export class LimitTestsImpl extends GPUTestBase {
     await this.expectValidationError(test, shouldError, msg);
   }
 
+  /**
+   * Test a method on GPURenderCommandsMixin or GPUBindingCommandsMixin
+   * The function will be called with the mixin.
+   */
+  async testGPURenderAndBindingCommandsMixin(
+    encoderType: RenderEncoderType,
+    fn: ({
+      mixin,
+      bindGroup,
+    }: {
+      mixin: GPURenderCommandsMixin & GPUBindingCommandsMixin;
+      bindGroup: GPUBindGroup;
+    }) => void,
+    shouldError: boolean,
+    msg = ''
+  ) {
+    const { mixin, prep, test, bindGroup } = this._getGPURenderCommandsMixin(encoderType);
```

Comment:
not important old code nit: private functions can be TypeScript `private` (or even JavaScript `#` private)